### ISSUE TITLE
Operator methods Can cause SwiftFormat parsing to fail

### ISSUE THREAD

**Issue by orchetect** [nicklockwood/SwiftFormat#1258]
SwiftFormat 0.49.16

I ran into an issue recently where declaring certain operator methods can cause SwiftFormat to fail parsing.

This occurs with either `auto` or `5.5` for `--swiftversion`.

In the given source:

```swift
struct Foo {
    public static func FOO <RHS>(lhs: inout Self, rhs: RHS) {
        // empty
    }
}
```

If `FOO` is replaced with any of the following:

- `<<`
- `<<=`
- `<<<`
- `<<<=`
- and so on

Then SwiftFormat fails with the following error:

```
error: Unexpected token ) at 2:59
```

It appears that two or more `<` characters in the method name seems to trip up SwiftFormat's parsing.

**Comment by nicklockwood** [comment]
@orchetect this is fixed already in 0.49.17 I believe

**Comment by orchetect** [comment]
It does appear resolved. Thank you sir.
